specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/SessionPools.json

```diff
@@ -0,0 +1,747 @@
+          "type": "boolean"
+        },
+        "jwtToken": {
+          "description": "JWT token.",
+          "type": "string"
+        }
+      }
+    },
+    "CustomContainerTemplate": {
+      "description": "Custom container configuration.",
+      "type": "object",
+      "properties": {
+        "registryCredentials": {
+          "description": "Private container registry credentials for containers used by the Sessions of the Session Pool.",
+          "$ref": "#/definitions/SessionRegistryCredentials"
+        },
+        "containers": {
+          "description": "List of container definitions for the Sessions of the Session Pool.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SessionContainer"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "ingress": {
+          "description": "SessionPool Ingress configuration.",
+          "$ref": "#/definitions/SessionIngress"
+        },
+        "sessionAuthConfiguration": {
+          "description": "Auth configuration of the Sessions in this Session Pool.",
+          "$ref": "#/definitions/SessionAuthConfiguration"
+        }
+      }
+    },
+    "SessionNetworkConfiguration": {
+      "description": "Custom container configuration.",
+      "type": "object",
+      "properties": {
+        "enableEgress": {
+          "description": "Whether to enable egress traffic for sessions.",
+          "type": "boolean"
+        }
+      }
+    },
+    "SessionPool": {
+      "description": "Container App Session Pool.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Container App Session Pool resource specific properties",
+          "type": "object",
+          "properties": {
+            "environmentId": {
+              "description": "Resource ID of the Session Pool's environment.",
+              "type": "string",
+              "format": "arm-id",
+              "x-ms-arm-id-details": {
+                "allowedResources": [
+                  {
+                    "type": "Microsoft.App/managedEnvironments"
+                  }
+                ]
+              },
+              "x-ms-mutability": [
+                "create",
+                "read"
+              ]
+            },
+            "containerType": {
+              "description": "The container type of the Sessions.",
+              "enum": [
+                "JupyterPython",
+                "FunctionsPython",
+                "CustomContainer"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "ContainerType",
+                "modelAsString": true
+              }
+            },
+            "poolManagementType": {
```

Comment:
Sticking to Kind or Type suffix would be better for consistency? I have Kind suffix in the session.

Reply:
Updated name

Reply:
So, you are proceeding with `Type` everywhere instead of `Kind` ?

Reply:
Is this being used as a discriminator? Considered using swagger discriminator notations? x-ms-discriminator-value etc